src/widget/wnumberduration.cpp

```diff
@@ -0,0 +1,71 @@
+#include "widget/wnumberduration.h"
+
+#include <QTime>
+
+#include "control/controlproxy.h"
+#include "moc_wnumberduration.cpp"
+#include "util/duration.h"
+
+WNumberDuration::WNumberDuration(QWidget* parent)
+        : WNumber(parent),
+          m_displayFormat(TrackTime::DisplayFormat::TRADITIONAL) {
+}
+
+void WNumberDuration::setup(const QDomNode& node, const SkinContext& context) {
+    WNumber::setup(node, context);
+
+    QString formatString = context.selectString(node, "TimeFormat");
+    if (formatString == "kilo_seconds") {
+        m_displayFormat = TrackTime::DisplayFormat::KILO_SECONDS;
+    } else if (formatString == "seconds_long") {
+        m_displayFormat = TrackTime::DisplayFormat::SECONDS_LONG;
+    } else if (formatString == "seconds") {
+        m_displayFormat = TrackTime::DisplayFormat::SECONDS;
+    } else if (formatString == "traditional") {
+        m_displayFormat = TrackTime::DisplayFormat::TRADITIONAL;
+    } else if (formatString == "traditional_coarse") {
+        m_displayFormat = TrackTime::DisplayFormat::TRADITIONAL_COARSE;
+    } else {
```

Comment:
You might want to use `SKIN_WARNING` macro to alert of a potentially unsupported format. I understand that `TimeFormat` may be an arbitrary string passed to `QTime::toString`. Perhaps you could use two attributes (e.g `TimeFormat` is an enum, including `custom` and `CustomFormat` allow you to define the arbitrary string, stored in `m_displayFormatString`. This way, it will help debugging cases where user accidentally mistype a pre-constructed enum.

Reply:
How about two properties `PredefinedFormat` (or `StandardFormat`/`DefaultFormat`) and `CustomFormat`)? And just use the one that is specified, and emit a warning if both are specified (and fallback to `CustomFormat`).

Reply:
Yes, that sounds perfect!